### PR TITLE
Add do-patch

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

**Comment by Cogitri** [comment]
I haven't followed the discussion in IRC, but how about do_prepare?

**Comment by maxice8** [comment]
On Tue, Feb 26, 2019 at 05:52:05AM -0800, Rasmus Thomsen wrote:
> I haven't followed the discussion in IRC, but how about do_prepare?
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/9189#issuecomment-467445694

We discussed that in IRC

**Comment by Gottox** [comment]
`do_prepare()` is a semantic noop, What is prepared? The build? The package? with patch it's pretty clear what this step does.

**Comment by maxice8** [comment]
On Tue, Feb 26, 2019 at 07:23:46AM -0800, Enno Boland wrote:
> `do_prepare()` is a semantic noop, What is prepared? The build? The package? with patch it's pretty clear what this step does.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/9189#issuecomment-467481299

by that logic do_patch excludes anything that isn't patching, what one assumes to be the source code.

File: Manual.md [hunk 1]
```diff
@@ -914,6 +917,12 @@ package is downloaded, compiled and installed.
 
 - `post_extract()` Actions to execute after `do_extract()`.
 
+- `pre_patch()` Actions to execute after `post_extract()`.
+
+- `do_patch()` if defined use it to prepare the build environment and run hooks to apply patches.
+
+- `post_patch()` Actions to execute after `do_patch()`.
+
 - `pre_configure()` Actions to execute after `post_extract()`.
```
Comment (Chocimier) [inline Manual.md]: after post_patch

**Comment by maxice8** [comment]
On Tue, Feb 26, 2019 at 07:30:47AM -0800, Piotr wrote:
> Chocimier commented on this pull request.
> 
> 
> 
> > @@ -914,6 +917,12 @@ package is downloaded, compiled and installed.
>  
>  - `post_extract()` Actions to execute after `do_extract()`.
>  
> +- `pre_patch()` Actions to execute after `post_extract()`.
> +
> +- `do_patch()` if defined use it to prepare the build environment and run hooks to apply patches.
> +
> +- `post_patch()` Actions to execute after `do_patch()`.
> +
>  - `pre_configure()` Actions to execute after `post_extract()`.
> 
> after post_patch
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/9189#pullrequestreview-208024831

Fixed, thanks

File: xbps-src [hunk 2]
```diff
@@ -726,7 +730,7 @@ case "$XBPS_TARGET" in
             remove_pkg_autodeps
         fi
         ;;
-    fetch|extract|build|check|configure|install|pkg)
+    fetch|extract|patch|configure|build|install|pkg)
```
Comment (AluisioASG) [inline xbps-src]: I guess dropping `check` here was unintentional?

**Comment by maxice8** [comment]
On Tue, Feb 26, 2019 at 08:50:29AM -0800, Aluísio Augusto Silva Gonçalves wrote:
> AluisioASG commented on this pull request.
> 
> 
> 
> > @@ -726,7 +730,7 @@ case "$XBPS_TARGET" in
>              remove_pkg_autodeps
>          fi
>          ;;
> -    fetch|extract|build|check|configure|install|pkg)
> +    fetch|extract|patch|configure|build|install|pkg)
> 
> I guess dropping `check` here was unintentional?
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/9189#pullrequestreview-208077607

yes

**Pull request merged**